internal/semver/semver.go

```diff
@@ -0,0 +1,70 @@
+package semver
+
+import (
+	"encoding/json"
+	"fmt"
+	"io"
+
+	"github.com/hashicorp/go-version"
+	"github.com/hashicorp/vault/api"
+
+	"github.com/hashicorp/terraform-provider-vault/internal/consts"
+)
+
+func getTargetVaultVersion(client *api.Client) (string, error) {
+	// This endpoint only returns data upon an HTTP request
+	// Reads using the Client do not return data
+	path := "/v1/sys/seal-status"
+	r := client.NewRequest("GET", path)
```

Comment:
It looks like there's a [SealStatus()/SealStatusWithContext() client function](https://github.com/hashicorp/vault/blob/8de51d4ae6006504227c064d19901830b9e20736/api/sys_seal.go#L8) in vault/api that may simplify some of this a bit.

Reply:
Thanks for pointing that one out! Definitely helped simplify a lot of the logic. I should've probably done more digging! :)